src/js/claims-status/components/appeals-v2/Timeline.jsx

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import moment from 'moment';
+
+import { getEventContent } from '../../utils/appeals-v2-helpers';
+import CurrentStatus from './CurrentStatus';
+
+/**
+ * Timeline is in charge of the past events and current status.
+ */
+class Timeline extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      expanded: false
+    };
+  }
+
+  getPastEvents = () => {
+    return this.props.events.map((event, index) => {
+      const { title, description, liClass } = getEventContent(event);
+      return (
+        <li key={index} role="presentation" className={`process-step ${liClass}`}>
+          <h3>{title || 'Title here'}</h3>
+          <span className="appeal-event-date">on {moment(event.date, 'YYYY-MM-DD').format('MMMM d, YYYY')}</span>
+          <p>{description}</p>
+        </li>
+      );
+    });
+  }
+
+  toggleExpanded = () => {
+    this.setState({ expanded: !this.state.expanded });
```

Comment:
state changes based on prev. state can get tricky since `setState` doesn't guarantee when state changes. For these types of state changes, React team recommends using the functional version - `setState((prevState, props) => ({ expanded: !prevState.expanded }));`
Ref: [State Updates May Be Asynchronous](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-may-be-asynchronous)

Reply:
Oooh, didn't know this was a thing. Good call!